### PR TITLE
Add indexes and fix properties of the String constructor

### PR DESCRIPTION
**Pull request by SBSTP**
This PR fixes the issues documented in #180.

### CODE REVIEW

File: source/tests/01-stdlib/string.js [hunk 1]
```diff
@@ -39,6 +39,7 @@
  *  SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
  * _________________________________________________________________________
  */
+ require('lib/test');
```
Comment (charmander) [inline source/tests/01-stdlib/string.js]: Should this space be here?

Comment (SBSTP) [inline source/tests/01-stdlib/string.js]: Nope! Fixing...

**Comment by SBSTP** [comment]
Also going to fix the code style.

**Comment by maximecb** [comment]
Thank you for the fix :)

**Pull request merged**